.github/renovate.json5

```diff
@@ -99,6 +106,31 @@
       "matchUpdateTypes": ["major", "minor"],
       "matchPackagePatterns": ["grafana/grafana"],
       "enabled": false
+    },
+    {
+      // Pin certain components to the current version.
+      "matchDatasources": ["docker"],
+      "matchUpdateTypes": ["major", "minor", "patch"],
+      "matchPackagePatterns": [
+        "coredns/coredns",
+        "calico/node",
+        "calico/cni",
+        "calico/kube-controllers",
+        "calico/pod2daemon-flexvol",
+        "k8scloudprovider/openstack-cloud-controller-manager",
+        "k8scloudprovider/cinder-csi-plugin",
+        "nginx",
+      ],
+      "enabled": false
+    },
+    {
+      "matchFileNames": ["config/images/images.yaml"],
+      "postUpgradeTasks": {
+        "commands": [
+          "bash -c \"sed -i `yq '(.images[] | select(.source == \\\"{{{depName}}}\\\") | key) as \\\$imagePos | (.images[\\\$imagePos].tags | length) as \\\$tagLength | .images[\\\$imagePos].tags[\\\$tagLength - 1] | line' config/images/images.yaml`'i\\\  - {{{currentVersion}}}' config/images/images.yaml\""
```

Comment:
Nit: Could you add a comment to explain what this rule is doing? 😄

Reply:
Added comment.